app/assets/v2/js/cart-ethereum-polygon.js

```diff
@@ -338,7 +344,67 @@ Vue.component('grantsCartEthereumPolygon', {
       // transaction before the approval txs are confirmed, because if the approval txs
       // are not confirmed then estimateGas will fail.
 
-      return 70000;
+      let networkId = appCart.$refs.cart.networkId;
+
+      if (networkId !== '80001' && networkId !== '137' && appCart.$refs.cart.chainId !== '1' ||
+          appCart.$refs.cart.standardCheckoutInitiated) {
+        return;
+      }
+
+      let gasLimit = 0;
+
+      // If user has enough balance within Polygon, cost equals the minimum amount
+      let { isBalanceSufficient, requiredAmounts } = await this.hasEnoughBalanceInPolygon();
+
+      if (!isBalanceSufficient) {
+        // If we're here, user needs at least one L1 deposit, so let's calculate the total cost
+        requiredAmounts = objectMap(requiredAmounts, value => {
+          if (value.isBalanceSufficient == false) {
+            return value.amount;
+          }
+        });
+
+        for (const tokenSymbol in requiredAmounts) {
+          if (tokenSymbol === 'ETH') {
+            gasLimit += 94659; // add ~94.66k gas for ETH deposits
+          } else {
+            gasLimit += 103000; // add 103k gas for token deposits
+          }
+        }
+      }
+
+      // If we have a cart where all donations are in Dai, we use a linear regression to
+      // estimate gas costs based on real checkout transaction data, and add a 50% margin
+      const donationCurrencies = this.donationInputs.map(donation => donation.token);
+      const daiAddress = this.getTokenByName('DAI')?.addr;
+      const isAllDai = donationCurrencies.every((addr) => addr === daiAddress);
+
+      if (isAllDai) {
+        if (donationCurrencies.length === 1) {
+          // Special case since we overestimate here otherwise
+          return gasLimit + 65000;
+        }
+        // https://github.com/mds1/Gitcoin-Checkout-Gas-Analysis
+        return gasLimit + 10000 * donationCurrencies.length + 80000;
```

Comment:
Similarly I think I saw a plotly dashboard link somewhere, where would handy to include those here